### PR TITLE
[usage] add optional UserID to filter usage

### PR DESCRIPTION
**Pull request by svenefftinge**
## Description

Filters usage data for members to only their own usage.

| Owner | Member |
| ------- | -------- |
| Screenshot 2023-08-07 at 16 37 52 | Screenshot 2023-08-07 at 16 37 44 |

## Related Issue(s)

Fixes EXP-357

## How to test

- Join this org ([invite link](https://se-usage-for-user.preview.gitpod-dev.com/orgs/join?inviteId=b597fe58-77f0-400e-b944-040a919ee30d)) (which already has usage)
- check no usage in the usage view
- start and stop a workspace
- wait for reconciliation and verify the usage record is there.

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - se-usage-for-user</li>
	<li><b>🔗 URL</b> - <a href="https://se-usage-for-user.preview.gitpod-dev.com/workspaces" target="_blank">se-usage-for-user.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - se-usage-for-user-gha.14830</li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

**Comment by svenefftinge** [comment]
@loujaybee @gtsiolis using the preview env you can try how the filtered usage view looks&feels. 
Here is an invite to my org, which has usage (see screenshots):
https://se-usage-for-user.preview.gitpod-dev.com/orgs/join?inviteId=b597fe58-77f0-400e-b944-040a919ee30d

As you see I have only changed the logic behind the API such that usage entries are filtered by userId when the requestor is a member. I believe we should also adjust the copy, ideally in a generic way such that we don't have two different copies depending on your role.

File: components/gitpod-protocol/src/usage.ts [hunk 1]
```diff
@@ -6,9 +6,10 @@
 
 import { WorkspaceType } from "./protocol";
 
-// types below are copied over from components/usage-api/typescript/src/usage/v1/usage_pb.d.ts
+// types below are manually kept in sycn with components/usage-api/typescript/src/usage/v1/usage_pb.d.ts
```
Comment (filiptronicek) [inline components/gitpod-protocol/src/usage.ts:9]:
```suggestion
// types below are manually kept in sync with components/usage-api/typescript/src/usage/v1/usage_pb.d.ts
```

**Review by filiptronicek — APPROVED** [review]
suggestion: when the user is indeed only looking at their own usage, can we omit the user details?

image

otherwise works as expected 🧡

**Review by akosyakov — APPROVED** [review]

**Comment by svenefftinge** [comment]
@loujaybee I'm merging this as is. We can do follow-up changes if necessary

/unhold

**Comment by loujaybee** [comment]
Ah, I'm just catching up on this now @svenefftinge, looks good. I agree that we need to look at copy changes / UX as otherwise it can be confusing why this view doesn't show all users. I can look at with @gtsiolis. 

Thank you !

**Pull request merged**